### ISSUE TITLE
Refactor naming, file and folder structure of editor configurations in contrast to existing lint rules

### ISSUE THREAD

**Issue by MrCube42** [typescript-eslint/tslint-to-eslint-config#275]
This is a follow-up issue of this PR: https://github.com/typescript-eslint/tslint-to-eslint-config/pull/250.
It is only valid if this branch gets merged.

Original discussion https://github.com/typescript-eslint/tslint-to-eslint-config/pull/250#discussion_r346635958:

> Reading through the branch, the difference between general linter config names (e.g. convertConfig) and their settings file equivalents (e.g. convertSettingsConfig) was a little hard to keep track of. That seems like something good for a followup issue+PR as well to clarify though, to keep the PR from getting too massive...

### 🚀 Feature Request

This refactoring should rather touch the general functionality. The goal is to clarify the namings, similarities and differences of rules/linter conversions and their editor settings/configurations equivalents. Currently there seems to be a great portion of copy-paste code. The similar functionality and namings make it hard to follow the different kinds of conversions. However it may involve a greater architectural change to clean this up.

#### Existing Behavior

- Basically `convertConfig.ts` and everything named `rules` converts TSLint rules into ESLint rules
- Similar `convertEditorConfig.ts`, everything named `editorConfig` or `settings` converts TSLint editor configuration settings into ESLint editor configuration settings

#### Change Proposal

- Clarify on the naming similarities and differences first: Do we talk about *lint rules* and *editor configurations* or *editor settings* or *editor configuration settings*.
- The naming and folder structure should make it easy and clear to follow the different parts.
